modules/frontend/metrics_query_handler.go

```diff
@@ -90,15 +97,18 @@ func newMetricsQueryInstantHTTPHandler(cfg Config, next pipeline.AsyncRoundTripp
 		}
 		start := time.Now()
 
+		if dataAccessController != nil {
+			if err := dataAccessController.HandleHTTPQueryInstantReq(req); err != nil {
+				level.Error(logger).Log("msg", "http instant query: access control handling failed", "err", err)
+				return httpInvalidRequest(err), nil
+			}
+		}
+
 		// Parse request
 		i, err := api.ParseQueryInstantRequest(req)
 		if err != nil {
 			level.Error(logger).Log("msg", "query instant: parse search request failed", "err", err)
-			return &http.Response{
-				StatusCode: http.StatusBadRequest,
-				Status:     http.StatusText(http.StatusBadRequest),
-				Body:       io.NopCloser(strings.NewReader(err.Error())),
-			}, nil
+			return httpInvalidRequest(err), nil
```

Comment:
Agree with Adrian's question about the mapping of error to http response.  Could there be other errors where we should return 500 internal server error?  gRPC should be ok, we can embed the needed response within the error like `grpcErr = status.Error(codes.InvalidArgument, <message>)`   It's ok to proceed and address this up later if we aren't sure.